weave/errors.py

```diff
@@ -1,9 +1,20 @@
+from typing import Optional
+
 class WeaveUnmergableArtifactsError(Exception):
     pass
 
 
 class WeaveBaseError(Exception):
-    pass
+    def __init__(self, message: str, fingerprint: Optional[str] = None) -> None:
+        super().__init__(message)
+        self.fingerprint = fingerprint
```

Comment:
please add a type def for fingerprint, also it should probably be an array to match the type that sentry expects

Reply:
This will require updating all other callers to use lists